iceoryx_posh/CMakeLists.txt

```diff
@@ -299,28 +299,14 @@ setup_install_directories_and_export_package(
     INCLUDE_DIRECTORY include/
 )
 
-if(ROUDI_ENVIRONMENT OR BUILD_TEST)
-    #
-    ######### posh roudi environment ##########
-    #
-    find_package(iceoryx_hoofs_testing REQUIRED)
-    iox_add_library(
-        STATIC
-        TARGET                  iceoryx_posh_testing
-        NAMESPACE               iceoryx_posh_testing
-        PROJECT_PREFIX          ${PREFIX}
-        BUILD_INTERFACE         ${CMAKE_CURRENT_SOURCE_DIR}/testing/include
-        INSTALL_INTERFACE       include/${PREFIX}
-        PRIVATE_LIBS            iceoryx_posh::iceoryx_posh
-                                iceoryx_hoofs::iceoryx_hoofs
-                                iceoryx_posh::iceoryx_posh_roudi
-                                iceoryx_hoofs_testing::iceoryx_hoofs_testing
-        FILES
-            testing/roudi_environment/runtime_test_interface.cpp
-            testing/roudi_environment/roudi_environment.cpp
-    )
+# Finding gtest and adding the subdirectories is split to support the use case of
+# building the testing lib without the tests by providing gtest externally
+if(NOT GTest_FOUND AND (ROUDI_ENVIRONMENT OR BUILD_TEST))
+    find_package(GTest CONFIG REQUIRED)
```

Comment:
Would it be an option to move this into `test/CMakeLists.txt`/`testing/CMakeLists.txt`?